app/assets/javascripts/helpers/form_helpers.js

```diff
@@ -0,0 +1,88 @@
+var FORM_HELPERS = (function (){
+  'use strict';
+
+  /**
+   * Finds the form fields in the given form.
+   *
+   * @param {jQuery} $form The form to search.
+   * @param {String|undefined} selector The filter to apply on the returned fields.
+   *
+   * @returns {jQuery} The set of fields matching the filter.
+   */
+  function findFormFields($form, selector) {
+    var $result = $form.find('textarea, input');
+    if (selector) {
+      $result = $result.filter(selector);
+    }
+
+    return $result;
+  };
+
+  /**
+   * Builds the form data from the given form.
+   *
+   * This is a less sophisticated version of $.serialize() in that it only supports inputs and
+   * textareas.
+   *
+   * @param {jQuery} $form The form being submitted.
+   * @returns {Object} The form data to be submitted.
+   */
+  function buildFormData ($form) {
+    var $fields = findFormFields($form, ':not(:disabled)');
+    var data = {
+      authenticity_token: $(document).find('meta[name="csrf-token"]').attr('content')
+    };
+    $fields.each(function() {
+      if (this.name === '') {
+        return;
+      }
+
+      data[this.name] = $(this).val();
+    });
+
+    return data;
+  };
+
+
+  /**
+   * @callback formSubmitSuccessCallback
+   * @param {data} data The server response
+   * @param {HTMLElement} form The submitted form
```

Comment:
OK then, seems you are correct from JS doc. But document this way a bit overkill to me ><